### PR TITLE
nRF5340: Add support for QDEC in device tree and Kconfig

### PR DESCRIPTION
**Pull request by ioannisg**
Adding support for QDEC in nRF5340

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

File: modules/hal_nordic/nrfx/Kconfig [hunk 1]
```diff
@@ -129,9 +129,19 @@ config NRFX_PWM3
...
+	bool "Enable QDEC0 instance"
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0
+	select NRFX_QDEC
+
+config NRFX_QDEC1
+	bool "Enable QDEC1 instance"
+	depends on HAS_HW_NRF_QDEC1
+	select NRFX_QDEC
+
+	config NRFX_QSPI
```
Comment (jfischer-no) [inline modules/hal_nordic/nrfx/Kconfig:144]: Looks like unintentional indentation.

**Review by jfischer-no — DISMISSED** [review]

File: modules/hal_nordic/nrfx/Kconfig (hunk 1 above)
Comment (ioannisg) [reply modules/hal_nordic/nrfx/Kconfig]: Fixed, thanks @jfischer-no

**Comment by ioannisg** [comment]
I'd need @anangl to review before we can merge that.

File: drivers/sensor/qdec_nrfx/Kconfig [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 config QDEC_NRFX
 	bool "Nordic QDEC nrfx driver"
-	depends on HAS_HW_NRF_QDEC
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0 || HAS_HW_NRF_QDEC1
```
Comment (anangl) [inline drivers/sensor/qdec_nrfx/Kconfig:6]: The nrfx_qdec driver, and consequently the qdec_nrfx shim for it, currently supports only single QDEC instance (for nRF53 it is `NRF_QDEC0`, which is mapped to `NRF_QDEC` in [nrfx_config_nrf5340_application.h](https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/hal_nordic/nrfx/nrfx_config_nrf5340_application.h#L101)), so I think `HAS_HW_NRF_QDEC1` should not be added here yet, to not bring confusion in this regard.

File: modules/hal_nordic/nrfx/Kconfig [hunk 3]
```diff
@@ -129,7 +129,17 @@ config NRFX_PWM3
...
-	depends on HAS_HW_NRF_QDEC
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0 || HAS_HW_NRF_QDEC1
+
+config NRFX_QDEC0
+	bool "Enable QDEC0 instance"
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0
+	select NRFX_QDEC
+
+config NRFX_QDEC1
+	bool "Enable QDEC1 instance"
```
Comment (anangl) [inline modules/hal_nordic/nrfx/Kconfig:140]: As I mentioned above, only `NRF_QDEC`/`NRF_QDEC0` is currently supported by the driver, so the user cannot actually select instances to be enabled. So at this point I think it's undesirable to introduce these two options.

File: modules/hal_nordic/nrfx/Kconfig [hunk 4]
```diff
@@ -129,7 +129,17 @@ config NRFX_PWM3
 
 config NRFX_QDEC
 	bool "Enable QDEC driver"
-	depends on HAS_HW_NRF_QDEC
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0 || HAS_HW_NRF_QDEC1
```
Comment (anangl) [inline modules/hal_nordic/nrfx/Kconfig:132]:
```suggestion
	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0
```

**Review by anangl — CHANGES_REQUESTED** [review]

File: modules/hal_nordic/nrfx/Kconfig (hunk 4 above)
Comment (ioannisg) [reply modules/hal_nordic/nrfx/Kconfig:132]: OK, so you're accepting to keep the HAS_HW_NRF_QDEC0 as "used" symbol here?

Comment (anangl) [reply modules/hal_nordic/nrfx/Kconfig]: yes

**Review by anangl — APPROVED** [review]

**Pull request merged**